### PR TITLE
Day 2 Lessons

### PR DESCRIPTION
**Pull request by pshriwise**
@ChristinaLK This is what I have planned for day 2 Python. I ended up breaking it into three sections, and I have plans to add some more content if we think its necessary but I have a feeling this will be enough and we can kind of open up the floor to questions afterward.

Some specific areas I'm hoping for feedback on: 

  - Is it ok to be doing all of this work in the `data` directory?
  - Should I be git-ignoring the Jupyter notebooks?
  - Does the whole thing flow well?
  - And, more specifically, does the importance of our refactor come across as strongly as I hope?

### CODE REVIEW

**Comment by pshriwise** [comment]
Hoping that @sstevens2 might have a few minutes to take a look as well if possible seeing as there is a fair amount of git work in here too!

**Comment by pshriwise** [comment]
One other thing I was hoping to get feedback on is whether or not I should break changes in the code of the refactor lesson into multiple commits or not. Right now we're making a pretty substantial change to the code in one commit and making a new branch for one commit. This seems a little odd to me. Thoughts on that would be appreciated as well!

File: _episodes/31-cmd-line.md [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+keypoints:
+- "The `sys` library connects a Python program to the system it is running on."
+---
+
+The Jupyter Notebook and other interactive tools are great for
+prototyping code and exploring data, but sooner or later one will
+want to use that code in a program we can run from the command line. 
+In order to do that, we need to make our programs work like other
+Unix command-line tools. For example, we may want a program that
+reads a data set and prints the average inflammation per patient.
```
Comment (ChristinaLK) [inline _episodes/31-cmd-line.md]: inflammation --> something else?

File: _episodes/31-cmd-line.md [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+>
+> The Jupyter Notebook has the ability to convert all of the cells of a
+> current Notebook into a python program. To do this, go to `File` -> `Download as`
+> and select `Python (.py)` to get the current notebook as a Python script.
+> 
+{: .callout}
+
+To ensure that we're all starting with the same set of code. Please
+copy the text below into a file called `gdp_plots.py` in our working directory
+with the gapminder data (`~/data`).
```
Comment (ChristinaLK) [inline _episodes/31-cmd-line.md]: This path will probably be `~/Desktop/python-novice-gapminder/data`

File: _episodes/31-cmd-line.md [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+Python module used to store information about the program and its running
+environment, including what arguments were passed to the program when the
+command was executed. These arguments are stored as a list in `sys.argv`.
+
+These arguments can be accessed in our program by importing the `sys`
+module. The first argument in `sys.argv` is always the name of the program, so
+we'll find any additional arguments right after that in the list.
+
+Let's try this out in a separate location on your machine. The directory above
+our current directory (a.k.a. the home directory) should do nicely
```
Comment (ChristinaLK) [inline _episodes/31-cmd-line.md:176]: may not be true...

**Review by ChristinaLK — COMMENTED** [review]
I think this looks good!  I love the defensive programming + refactoring examples -- I feel like that's a really good example of what it means to write healthy code.

File: _episodes/31-cmd-line.md (hunk 3 above)
Comment (pshriwise) [reply _episodes/31-cmd-line.md:176]: Hrmmm I thought about having them just add this test program to the repo, but I'm also trying not to spend too much time on git. Would that be better?

File: _episodes/31-cmd-line.md (hunk 1 above)
Comment (pshriwise) [reply _episodes/31-cmd-line.md]: eep! Thanks!

File: _episodes/31-cmd-line.md (hunk 3 above)
Comment (ChristinaLK) [reply _episodes/31-cmd-line.md:176]: the idea is fine -- their home directory may just be one folder above.  But if the git repo is in the `data` folder, then just the level above should be fine.

**Pull request merged**